src/modules/ChannelList/dux/reducers.ts

```diff
@@ -108,22 +109,61 @@ export default function channelListReducer(
       .with({ type: channelListActions.ON_USER_LEFT }, (action) => {
         const { channel, isMe } = action.payload;
         const { allChannels, currentUserId, currentChannel, channelListQuery, disableAutoSelect } = state;
-        let nextChannels = allChannels.filter((ch) => ch.url !== channel.url);
-        let nextChannel = null;
-        if (channelListQuery) {
-          if (filterChannelListParams(channelListQuery, channel, currentUserId)) {
-            // Good to [add to/keep in] the ChannelList
+        let nextChannels = [...allChannels];
+        let nextChannel: GroupChannel = channel;
+
+        /**
+         * If current channel list includes the channel, update the channel list.
+         *
+         * 1. If I left channel:
+         *   - Remove the channel from channel list
+         *   - Replace currentChannel with the next ordered channel
+         * 2. If other member left channel:
+         *   2-1. If query is given:
+         *     2-1-1. If channel no longer matches the query
+         *       - Remove the channel from channel list
+         *       - Replace currentChannel with the next ordered channel
+         *     2-1-2. If channel matches the query:
+         *       - Upsert channel list with the channel
+         *       - Replace currentChannel IFF url is same
+         *   2-2. If query is not given,
+         *     - Upsert channel list with the channel
+         *     - Replace currentChannel IFF url is same
+         */
```

Comment:
`1` and `2-1-1`, `2-1-2` and `2-2` have same action, can we reduce the lines?

Reply:
yes it is done!